app/assets/javascripts/components/reorderable-list.js

```diff
@@ -0,0 +1,84 @@
+//= require sortablejs/Sortable.js
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function ReorderableList () { }
+
+  ReorderableList.prototype.start = function ($module) {
+    this.$module = $module[0]
+    this.$upButtons = this.$module.querySelectorAll('.app-c-reorderable-list__actions button:first-of-type')
+    this.$downButtons = this.$module.querySelectorAll('.app-c-reorderable-list__actions button:last-of-type')
+
+    this.sortable = new window.Sortable.create(this.$module, { // eslint-disable-line new-cap
+      chosenClass: 'app-c-reorderable-list__item--chosen',
+      dragClass: 'app-c-reorderable-list__item--drag',
+      onSort: this.updateOrderIndexes.bind(this)
+    })
+
+    if (typeof window.matchMedia === 'function') {
+      this.setupResponsiveChecks()
+    } else {
+      this.sortable.option('disabled', true)
+    }
+
+    var boundOnClickUpButton = this.onClickUpButton.bind(this)
+    this.$upButtons.forEach(function (button) {
+      button.addEventListener('click', boundOnClickUpButton)
+    })
+
+    var boundOnClickDownButton = this.onClickDownButton.bind(this)
+    this.$downButtons.forEach(function (button) {
+      button.addEventListener('click', boundOnClickDownButton)
+    })
+  }
+
+  ReorderableList.prototype.setupResponsiveChecks = function () {
+    this.mql = window.matchMedia('(min-width: 40.0625em)')
+    this.mql.addListener(this.checkMode.bind(this))
+    this.checkMode()
+  }
+
+  ReorderableList.prototype.checkMode = function () {
+    if (this.mql.matches) {
+      this.sortable.option('disabled', false)
+    } else {
+      this.sortable.option('disabled', true)
+    }
+  }
+
+  ReorderableList.prototype.onClickUpButton = function (e) {
+    var item = e.target.closest('.app-c-reorderable-list__item')
+    var previousItem = item.previousElementSibling
+    if (item && previousItem) {
+      item.parentNode.insertBefore(item, previousItem)
+      this.updateOrderIndexes()
+    }
+    // if triggered by keyboard preserve focus on button
+    if (e.detail === 0) {
+      e.target.focus()
```

Comment:
There seems to be a case here where if you move an item up to the top with keyboard this will apply a focus to a hidden element - and after that the tab index seems a little strange. Same situation happens on down too.

Reply:
Yes, it's an edge case that I haven't captured in the code. We will need to add more logic to cope with these two cases (first visible up button and last visible down button).